### PR TITLE
adding new tool "cppcheck"

### LINKED ISSUE #5217: Add CppCheck static-analysis tool
@mrodozov @tulamor can you please add https://github.com/danmar/cppcheck to cmssw externals. Its a static-analysis tool for C/C++ code

### PR DESCRIPTION
**Pull request by tulamor**
Resolves: #5217

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: cppcheck.spec [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+BuildRequires: cmake gmake
+Requires: pcre
+
+%prep
+%setup -n %{n}-%{realversion}
+ 
+%build
+cmake . \
+  -DCMAKE_INSTALL_PREFIX:PATH="%{i}" \
```
Comment (smuzaffar) [inline cppcheck.spec:12]: @tulamor , please use -DCMAKE_BUILD_TYPE=Release to build release mode (default is debug).

File: cppcheck.spec [hunk 2]
```diff
@@ -0,0 +1,22 @@
+### RPM external cppcheck 1.89
+Source: https://github.com/danmar/cppcheck/archive/%{realversion}.tar.gz
+
+BuildRequires: cmake gmake
+Requires: pcre
```
Comment (smuzaffar) [inline cppcheck.spec]: please add python dependency as it uses system python

https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5583a2/2448/external/cppcheck/1.89-5583a2/log
```
-- Found PythonInterp: /usr/bin/python (found version "2.7.5") 
```

File: cppcheck.spec [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+%prep
+%setup -n %{n}-%{realversion}
+ 
+%build
+cmake . \
+  -DCMAKE_INSTALL_PREFIX:PATH="%{i}" \
+  -DBUILD_GUI=OFF \
+  -DHAVE_RULE=ON \
```
Comment (smuzaffar) [inline cppcheck.spec]: this should be -DHAVE_RULES=OFF , in build logs I see
```
-- 
-- HAVE_RULES =            OFF
-- 
-- Change a value with: cmake -D<Variable>=<Value>
-- 
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    HAVE_RULE
```

**Comment by mrodozov** [comment]
please test

**Comment by smuzaffar** [comment]
-1
this tool is not relocatable and it is not trivial to patch it. We will open a issue with developers to make it relocatable

**Pull request closed without merging**